cpp/include/logistic_regression.h

```diff
@@ -69,18 +69,12 @@ class LogisticRegression {
   /// returns it as a vector. 
   ///
   /// 
-  static Vector<T> Fit(const Matrix<T> &xin, const Vector<T> &y, int iterations, T alpha){
-    Matrix<T> x;
+  static Vector<T> Fit( Matrix<T> &x, const Vector<T> &y, int iterations, T alpha){
     Vector<T> gradient, thetas;
 
-    x.resize(xin.rows(), xin.cols() + 1);
+    x.conservativeResize(x.rows(), x.cols() + 1);
```

Comment:
Using conservativeResize and then shift n * d data over works, but is very costly. It would actually be interesting to test the time spent on shifting vs the time spent on the iteration. Say let's fix d = 10, and then use n = 100, 1000, 10k, 100k, and 1 for input matrix X and plot two times spent on data shifting and theta updating. We could see some interesting result. 

We do not need to really add a column to X and an element to theta, the reason the coursera class did it is for just for algorithm simplicity but at the cost of implementation. I will find some materials that show you how to do this without the appending step. But in the mean time, could you add time stamps and try data in different sizes?